src/migrate.ts

```diff
@@ -42,55 +42,94 @@ export async function migrate (args: ArgumentsCamelCase<Params>): Promise<void>
   }
 
   const numKeys = await sbp('chelonia.db/keyCount')
+  let numMigratedKeys = 0
   let numVisitedKeys = 0
-  let shouldExit = 0
-
-  const handleSignal = (signal: string, code: number) => {
-    process.on(signal, () => {
-      console.error(`Received signal ${signal} (${code}). Finishing current operation.`)
-      // Exit codes follow the 128 + signal code convention.
-      // See <https://tldp.org/LDP/abs/html/exitcodes.html>
-      shouldExit = 128 + code
-    })
+
+  const reportStatus = () => {
+    console.log(`[chel] ${colors.green('Migrated:')} ${numMigratedKeys} entries`)
   }
 
-  // Codes from <signal.h>
-  ;([
-    ['SIGHUP', 1],
-    ['SIGINT', 2],
-    ['SIGQUIT', 3],
-    ['SIGTERM', 15],
-    ['SIGUSR1', 10],
-    ['SIGUSR2', 11]
-  ] as [string, number][]).forEach(([signal, code]) => handleSignal(signal, code))
+  const checkAndExit = (() => {
+    let interruptCount = 0
+    let shouldExit = 0
+
+    const handleSignal = (signal: string, code: number) => {
+      process.on(signal, () => {
+        // Exit codes follow the 128 + signal code convention.
+        // See <https://tldp.org/LDP/abs/html/exitcodes.html>
+        shouldExit = 128 + code
+
+        if (++interruptCount < 3) {
+          console.error(`Received signal ${signal} (${code}). Finishing current operation.`)
+        } else {
+          console.error(`Received signal ${signal} (${code}). Force quitting.`)
+          reportStatus()
+          exit(shouldExit)
+        }
+      })
+    }
+
+    const checkAndExit = async () => {
+      if (shouldExit) {
+        await backendTo.close()
+        reportStatus()
+        exit(shouldExit)
+      }
+    }
+
+    // Codes from <signal.h>
+    ;([
+      ['SIGHUP', 1],
+      ['SIGINT', 2],
+      ['SIGQUIT', 3],
+      ['SIGTERM', 15],
+      ['SIGUSR1', 10],
+      ['SIGUSR2', 11]
+    ] as [string, number][]).forEach(([signal, code]) => handleSignal(signal, code))
+
+    return checkAndExit
+  })()
 
   let lastReportedPercentage = 0
   for await (const key of sbp('chelonia.db/iterKeys')) {
-    if (!isValidKey(key)) continue
-    const value = await sbp('chelonia.db/get', key)
-    if (shouldExit) {
-      exit(shouldExit)
+    numVisitedKeys++
+    if (!isValidKey(key)) {
+      console.debug('[chel] Skipping invalid key', key)
+      continue
+    }
+    // `any:` prefix needed to get the raw value, else the default is getting
+    // a string, which will be encoded as UTF-8. This can cause data loss.
+    let value: unknown
+    try {
+      value = await sbp('chelonia.db/get', `any:${key}`)
+    } catch (e) {
+      reportStatus()
+      console.error('Error reading from source databse', key, e)
```

Comment:
There's a typo here and below in `databse`.

Also, what about making the key a bit clearer in the log message?

E.g. something like:

```ts
console.error(`Error reading from source database key '${key}':`, e)
```